gotocompany/shield/v1beta1/servicedata.proto

```diff
@@ -0,0 +1,98 @@
+syntax = "proto3";
+
+package gotocompany.shield.v1beta1;
+
+import "google/api/annotations.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+import "validate/validate.proto";
+
+option go_package = "github.com/goto/proton/shield/v1;shieldv1beta1";
+option java_outer_classname = "ServiceData";
+option java_package = "com.gotocompany.proton.shield.v1beta1";
+// These annotations are used when generating the OpenAPI file.
+option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+  info: {
+    title: "Service Data",
+    version: "0.1.0";
+  };
+  schemes: HTTP;
+};
+
+service ServiceDataService {
+   // Service Data
+   rpc CreateServiceDataKey(CreateServiceDataKeyRequest) returns (CreateServiceDataKeyResponse) {
+    option (google.api.http) = {
+      post: "/v1beta1/servicedata",
+      body: "body"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "Service Data";
+      summary: "Create Service Data Key";
+    };
+  }
+
+  rpc UpsertUserServiceData(UpsertUserServiceDataRequest) returns (UpsertUserServiceDataResponse) {
+    option (google.api.http) = {
+      put: "/v1beta1/users/{id}/servicedata",
+      body: "body"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "Service Data";
+      summary: "Upsert User Service Data";
+    };
+  }
+    
+  rpc UpsertGroupServiceData(UpsertGroupServiceDataRequest) returns (UpsertGroupServiceDataResponse) {
+    option (google.api.http) = {
+      put: "/v1beta1/groups/{id}/servicedata",
+      body: "body"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      tags: "Service Data";
+      summary: "Upsert Group Service Data";
+    };
+  }
+}
+
+message ServiceDataKeyRequestBody {
+  string project = 1;
+  string key = 2;
+  string description = 3;
+}
+
+message ServiceDataKey {
+  string urn = 1;
+  string id = 2;
+}
+
+message CreateServiceDataKeyRequest {
+  ServiceDataKeyRequestBody body = 1;
```

Comment:
can we just explicitly listing down all fields here?
```suggestion
  string project = 1;
  string key = 2;
  string description = 3;
```

Reply:
this is needed isn't it, if not it will be marked as query param?

Reply:
Actually it depends on the rpc that uses this message.

Can you please verify if havinf this explicit body is being used widely in shield? If yes we can keep it as it is for consistency

Reply:
it is. also the entity id is being sent in param so shouldn't it be separated?

Reply:
right, if `body: "*"`, it is okay to combine it.
but since Shield has widely use body, we can keep it as it is to make the implementation consistent.